core/src/smartcontracts/wasm.rs

```diff
@@ -763,39 +870,61 @@ impl<S: state::LimitsMut> Runtime<S> {
         let dealloc_fn =
             Self::get_typed_func(&instance, &mut store, import::SMART_CONTRACT_DEALLOC)
                 .expect("Checked at instantiation step");
-        codec::decode_with_length_prefix_from_memory(&memory, &dealloc_fn, &mut store, offset)
-            .map_err(Error::Decode)
+        let validation_res =
+            codec::decode_with_length_prefix_from_memory(&memory, &dealloc_fn, &mut store, offset)
+                .map_err(Error::Decode)?;
+
+        let mut state = store.into_data();
+        let executed_queries = state.take_executed_queries();
+        forget_all_executed_queries(state.wsv().query_handle(), executed_queries)?;
+        Ok(validation_res)
     }
 }
 
 #[allow(clippy::needless_pass_by_value)]
-impl<S: state::Authority + state::Wsv + state::WsvMut> Runtime<S> {
+impl<S: state::Authority + state::Wsv + state::WsvMut + state::AddExecutedQuery> Runtime<S> {
```

Comment:
```suggestion
impl<S: state::Authority + state::WsvMut + state::AddExecutedQuery> Runtime<S> {
```

shouldn't `WsvMut` imply `Wsv`?

Reply:
Fixed